Multisig/UI/App/UIKit/TransactionDetailsViewController.swift

```diff
@@ -45,48 +47,112 @@ class TransactionDetailsViewController: LoadableViewController, UITableViewDataS
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        title = "Transaction Details"
+
         builder = TransactionDetailCellBuilder(vc: self, tableView: tableView)
 
         tableView.delegate = self
         tableView.dataSource = self
 
         tableView.rowHeight = UITableView.automaticDimension
         tableView.estimatedRowHeight = 48
+
+        configureConfirmButton()
+
+        notificationCenter.addObserver(
+            self, selector: #selector(lazyReloadData), name: .ownerKeyRemoved, object: nil)
+        notificationCenter.addObserver(
+            self, selector: #selector(lazyReloadData), name: .ownerKeyImported, object: nil)
     }
 
-    override func reloadData() {
-        super.reloadData()
-        reloadDataTask?.cancel()
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        trackEvent(.transactionsDetails)
+    }
+
+    // MARK: - Signing
+
+    fileprivate func configureConfirmButton() {
+        // confirm button sticks to the bottom of the screen
+        // and is on top of the table view.
+        // it is shown only when table view is shown.
+
+        confirmButton = UIButton(type: .custom)
+        confirmButton.setText("Confirm", .filled)
+        confirmButton.addTarget(self, action: #selector(didTapConfirm), for: .touchUpInside)
+        confirmButton.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(confirmButton)
+        NSLayoutConstraint.activate([
+            confirmButton.heightAnchor.constraint(equalToConstant: 56),
+            confirmButton.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -8),
+            confirmButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+            confirmButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16)
+        ])
+    }
+
+    override func showOnly(view: UIView) {
+        super.showOnly(view: view)
+        confirmButton.isHidden = view !== tableView || !showsConfirmButton
+    }
+
+    private var showsConfirmButton: Bool  {
+        App.configuration.toggles.signing && tx?.txStatus == .awaitingYourConfirmation
+    }
 
-        let loadingCompletion: (Result<SCG.TransactionDetails, Error>) -> Void = { [weak self] result in
-            guard let `self` = self else { return }
-            switch result {
-            case .failure(let error):
-                DispatchQueue.main.async { [weak self] in
-                    guard let `self` = self else { return }
-                    // ignore cancellation error due to cancelling the
-                    // currently running task. Otherwise user will see
-                    // meaningless message.
-                    if (error as NSError).code == URLError.cancelled.rawValue &&
-                        (error as NSError).domain == NSURLErrorDomain {
-                        return
+    @objc private func didTapConfirm() {
```

Comment:
I would like to have something like AlertManager so we can use all over the app, by just pass the required strings and call "Show" method

Reply:
please create an additional ticket for that 😄